index.js

```diff
@@ -1,18 +1,131 @@
+  let pgmTries = 1; 
+  let userRespYorN = await ask('? (Please type a Y or a N only) -> ');
+    let userRespYorNUC = userRespYorN.toUpperCase().trim();  // to capitalize letters and also trim spaces
+    
+  if (userRespYorNUC === 'Y') {
+    console.log('You stated that this was the correct number, Yea!');  
+    console.log('The program has now tried: ' + pgmTries + ' times.'); // game exits since it's done
+    process.exit();
+  } else if (userRespYorNUC === 'N') {
+    console.log('You stated it was NOT correct, or ' + userRespYorNUC + ' , Sorry.');
+    // keepGuessing
+
+  pgmTries = pgmTries + 1;  // add another try
+  console.log('The program will try another time - This time makes: ' + pgmTries + ' times.');
+  console.log('The maximum number of tries this program should theoretically take is only: ' + maxTries);
+
+  // it was a N, then continue - 
+  if (userRespYorNUC === 'N') { 
+    continueTrying = (true);}    
+  let userRespHorL = await ask("Is it higher (H) or lower (L)? ");
+    
+  // in case the user used lower case - 
+  let userRespHorLUC = userRespHorL.toUpperCase();  
+    
+  // goes to High/Low Logic -  
+    if (userRespHorLUC == 'H') {  // goes to Higher logic 
+   (gotoHighLoop = (true)); 
+     // get a Higher range --
+     let newMin = (cpuGuessNum + 1); // causes it to not include cur low guess
+     let newRange = (max - newMin);
+     cpuGuessNum = (newRange/2);
+  } else if (userRespHorLUC == 'L') {  // goes to Lower Logic
+    (gotoLowLoop = (true));
+    // get a lower range --
+     let newMax = (cpuGuessNum - 1); // causes it to no include cur high guess 
+     let newRange = (newMax - min);
+     cpuGuessNum = (newRange/2);
+     
+    // else needs to go to end of pgm after - 
+  } else {
+   cheatMsgUnexpectedErr(); // putting this in for messages that don't equal y or n
+  }
+   
+} else {
+  cheatMsg1();
+}  
+
+} else { 
+  cheatMsgRangeErr();
+}  
+
+endGameMsg(); // totally ends out of the game
+process.exit();
+}
+
+// -----------putting various message functions and exit functions below this line -----------
+
+function cheatMsg1() {
+  console.log('Sorry, something did not compute!');
+  endGameMsg();
   process.exit();
 }
+
+/* function cheatMsg2() {
+  console.log('Sorry, something did not compute! You entered: ' + userRespHorL + 'or we tried to compute: ' + userReponseHorLUC);
+} */  // commenting out for now until more code is written
+
+function endGameMsg() {
+  console.log("We're done then. Goodbye."); // temporarily ending game after one random try
+}
+
+function cheatMsgRangeErr() {
+   console.log('Sorry, the program is exiting because the number was outside the given range expected, Sorry.');
+   endGameMsg();
+   process.exit();
+}
+
+function cheatMsgUnexpectedErr() {
+  console.log('Sorry, the program is exiting because the answer given was not what was expected, Sorry.');
+  endGameMsg();
+  process.exit();
+} 
```

Comment:
For the cheat detector, if you remember the section in the [project description](https://bootcamp.burlingtoncodeacademy.com/projects/guess#cheat-detector), it has some fairly telling hints on how to attack it. It says you shouldn't be able to contradict a previous response, i.e. "But you said it was lower than 25, so it can't also be higher than 24!"

So, in saying this, think about how that might be translated into JavaScript. What if you were to say...
```js
if (min >= Math.floor(min + max) / 2)
```
or vice versa?